paasta_tools/api/api.py

```diff
@@ -110,27 +124,21 @@ def setup_paasta_api():
 
 
 def main(argv=None):
-    monkey.patch_all()
     args = parse_paasta_api_args()
+
     if args.debug:
-        logging.basicConfig(level=logging.DEBUG)
-    else:
-        logging.basicConfig(level=logging.WARNING)
+        os.environ["PAASTA_API_DEBUG"] = "1"
 
     if args.soa_dir:
-        settings.soa_dir = args.soa_dir
-
-    server = WSGIServer(('', int(args.port)), make_app())
-    log.info("paasta-api started on port %d with soa_dir %s" % (args.port, settings.soa_dir))
-
-    try:
-        # We create the Zookeeper pool here to prevent the context manager
```

Comment:
How confident are you that we don't need to do this somewhere else? I guess each worker will instantiate its own ZookeeperPool object but can each worker serve multiple requests in parallel?

Reply:
Fairly confident. We're running gunicorn with multiple workers but each worker only has one thread, and isn't using gevent or asyncio or anything like that to be able to do more than one request at a time.